Code/RDGeneral/LocaleSwitcher.h

```diff
@@ -19,8 +24,67 @@ namespace Utils {
 // instead of whatever we started in.
 class LocaleSwitcher {
  public:
-  LocaleSwitcher() { std::setlocale(LC_ALL, "C"); }
-  ~LocaleSwitcher() { std::setlocale(LC_ALL, ""); }
+#ifdef _MSC_VER  
+
+  LocaleSwitcher() {
+    if (!Recurse()) {
+#ifdef RDK_THREADSAFE_SSS
```

Comment:
We need to be careful about introducing these config options in header files that may be included in client code (because they would then need to know the compile settings of the library in order to build their code). 

Does this need to be in the header or can we create a .cpp file for the LocalSwitcher and handle it there?

I know you have an idea of fixing this via a central config header file that can be included, but until that's in place we need to be careful (and it's probably not a bad idea to limit this anyway).